pymongo/database.py

```diff
@@ -911,20 +985,38 @@ def drop_collection(
 
         if not isinstance(name, str):
             raise TypeError("name_or_collection must be an instance of str")
-
-        command = SON([("drop", name)])
-        if comment is not None:
-            command["comment"] = comment
-
-        with self.__client._socket_for_writes(session) as sock_info:
-            return self._command(
-                sock_info,
-                command,
-                allowable_errors=["ns not found", 26],
-                write_concern=self._write_concern_for(session),
-                parse_write_concern_error=True,
+        full_name = "%s.%s" % (self.name, name)
+        if (
+            not encrypted_fields
+            and self.client.options.auto_encryption_opts
+            and self.client.options.auto_encryption_opts._encrypted_fields_map
+        ):
+            encrypted_fields = self.client.options.auto_encryption_opts._encrypted_fields_map.get(
+                full_name
+            )
+        if not encrypted_fields and self.client.options.auto_encryption_opts:
```

Comment:
Were we going to factor this logic out to a helper method?

Reply:
The logic is slightly different for create_collection and drop_collection, so I'm not sure it makes sense to factor it out

Reply:
The large majority of the logic is identical and we can refactor it but we can address that in a future change.